theano/tensor/signal/pool.py

```diff
@@ -745,14 +817,22 @@ def c_code(self, node, name, inp, out, sub):
             PyErr_SetString(PyExc_ValueError, "gz must be a 4d ndarray");
             %(fail)s;
         }
+        // Getting ws, stride and pad
+        int ws0, ws1, st0, st1, pd0, pd1;
+        ws0 = *((npy_intp*)PyArray_GETPTR1(%(ws)s, 0));
+        ws1 = *((npy_intp*)PyArray_GETPTR1(%(ws)s, 1));
+        st0 = *((npy_intp*)PyArray_GETPTR1(%(stride)s, 0));
+        st1 = *((npy_intp*)PyArray_GETPTR1(%(stride)s, 1));
+        pd0 = *((npy_intp*)PyArray_GETPTR1(%(pad)s, 0));
+        pd1 = *((npy_intp*)PyArray_GETPTR1(%(pad)s, 1));
```

Comment:
assert the shape. (not the number of dimensions, this is done automatically for the c code, but not for the perform.